fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneOnlineIndexingTest.java

```diff
@@ -145,6 +153,115 @@ void luceneOnlineIndexingTestSimple() {
         assertTrue(allFiles.length < 12);
     }
 
+    @Test
+    void luceneOnlineIndexingTestWithRecordUpdates() throws IOException {
+        final Map<String, String> options = Map.of(
+                INDEX_PARTITION_BY_FIELD_NAME, "timestamp",
+                PRIMARY_KEY_SEGMENT_INDEX_V2_ENABLED, "true",
+                INDEX_PARTITION_HIGH_WATERMARK, String.valueOf(2));
+
+        Index index = complexPartitionedIndex(options);
+        disableIndex(index, COMPLEX_DOC);
+
+        final RecordLayerPropertyStorage contextProps = RecordLayerPropertyStorage.newBuilder()
+                .addProp(LuceneRecordContextProperties.LUCENE_REPARTITION_DOCUMENT_COUNT, 2)
+                .build();
+
+        long startTime = System.currentTimeMillis();
+        try (final FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            recordStore.saveRecord(createComplexDocument(1623L, WAYLON, 1, startTime));
+            recordStore.saveRecord(createComplexDocument(1547L, WAYLON, 1, startTime + 1000));
+            recordStore.saveRecord(createComplexDocument(2222L, WAYLON + " who?", 1, startTime + 2000));
+            recordStore.saveRecord(createComplexDocument(899L, ENGINEER_JOKE, 1, startTime + 3000));
+            context.commit();
+        }
+
+        // do not index any records and leave the index writeOnly
+        try (final FDBRecordContext context = openContext(contextProps)) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            final RuntimeException stopBuildException = new RuntimeException("stop build");
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .setInitialLimit(1)
+                    .setConfigLoader(config -> {
+                        throw stopBuildException;
+                    })
+                    .build()) {
+                assertTrue(recordStore.isIndexDisabled(index));
+                final RuntimeException thrown = assertThrows(RuntimeException.class, () -> indexBuilder.buildIndex(true));
+                assertSame(stopBuildException, thrown);
+            }
+        }
+
+        final LuceneIndexTestValidator luceneIndexTestValidator = new LuceneIndexTestValidator(() -> openContext(contextProps), context -> {
+            rebuildIndexMetaData(context, COMPLEX_DOC, index);
+            return recordStore;
+        });
+
+        final List<LucenePartitionInfoProto.LucenePartitionInfo> partitionMeta = luceneIndexTestValidator.getPartitionMeta(index, Tuple.from(1L));
+        assertTrue(partitionMeta.isEmpty());
+
+        // update a record; no partition information exist yet
```

Comment:
You also need to test updating a record that has already been inserted into the index, i.e. updating 1623L a second time.